src/prefect/_internal/compatibility/deprecated.py

```diff
@@ -272,3 +273,55 @@ def callback(_):
     DEPRECATED_MODULE_ALIASES.append(
         AliasedModuleDefinition(old_name, new_name, callback)
     )
+
+
+class AsyncCompatProxy(wrapt.ObjectProxy):
+    """
+    A proxy object that allows for awaiting a method that is no longer async.
+
+    See https://wrapt.readthedocs.io/en/master/wrappers.html#object-proxy for more
+    """
+
+    def __init__(self, wrapped, class_name: str, method_name: str):
+        super().__init__(wrapped)
+        self._self_class_name = class_name
+        self._self_method_name = method_name
+        self._self_already_awaited = False
```

Comment:
the `_self_` prefix is required to avoid attempts to set these things on the wrapped object, since we only want it on the proxy object
```suggestion
        self._self_class_name = class_name
        self._self_method_name = method_name
        self._self_already_awaited = False
```